src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -672,12 +746,9 @@ public class [|C|]
         public int P { get; init; }
         public bool B { get; init; }
 
-        public static bool operator ==(C c1, C c2) {
-            return c1.Equals(c2);
-        }
-
-        public static bool operator !=(C c1, C c2) {
-            return !(c1 == c2);
+        public override bool Equals(object? other)
+        {
+            return !(other is not C otherC || otherC.P != P || otherC.B != B);
```

Comment:
Not sure this is code I've ever seen before -- did this require explicit support or this is just fallout from the other things that were needed?

Reply:
The boolean inversion in general was made to support situations where we'd see an if statement that returns false if its requirements are satisfied, like this:
```
if (other is not C otherC *(other potential checks that would indicate inequality)*)
{
    return false;
}
return otherC.B == B && otherC.P == P;
```
Mainly, I wanted to make sure our comparisons were not only testing for equality but actually making sure each equality condition was met before we return true. Checking code like this sort of falls out of that, I only really needed to add support for the not operator.